src/Database.tsx

```diff
@@ -0,0 +1,174 @@
+import { Program } from './Programs';
+
+const defaultExerciseDatabase = [
+  { value: 'Bench', text: 'Bench' },
+  { value: 'Deadlift', text: 'Deadlift' },
+  { value: 'Squat', text: 'Squat' },
+];
+const defaultUserSettings = {
+  weightUnit: 'kg',
+  heightUnit: 'cm',
+};
+const defaultUserPRs = {
+  Bench: { tested1RM: 0, estimated1RM: 0 },
+  Deadlift: { tested1RM: 0, estimated1RM: 0 },
+  Squat: { tested1RM: 0, estimated1RM: 0 },
+  Total: { tested1RM: 0, estimated1RM: 0 },
+};
+
+const defaultPrograms = {
+  'Starting Strength': {
+    name: 'Starting Strength',
+    author: 'Mark Rippetoe',
+    level: 'Beginner',
+    days: '3',
+    program: [
+      [
+        { exerciseName: 'Squat', Sets: 3, Reps: 5 },
+        { exerciseName: 'Bench', Sets: 3, Reps: 5 },
+        { exerciseName: 'Deadlift', Sets: 1, Reps: 5 },
+      ],
+      [
+        { exerciseName: 'Squat', Sets: 3, Reps: 5 },
+        { exerciseName: 'Press', Sets: 3, Reps: 5 },
+        { exerciseName: 'Deadlift', Sets: 1, Reps: 5 },
+      ],
+    ],
+    notes: 'The goal is to add as much weight per session as possible. Eventually you can switch out deadlifts for rows and do those every other day.',
+  },
+  'Stronglifts 5x5': {
+    name: 'Stronglifts 5x5',
+    author: 'Mehdi',
+    level: 'Beginner',
+    days: '3',
+    program: [
+      [
+        { exerciseName: 'Squat', Sets: 5, Reps: 5 },
+        { exerciseName: 'Bench', Sets: 5, Reps: 5 },
+        { exerciseName: 'Barbell Row', Sets: 5, Reps: 5 },
+      ],
+      [
+        { exerciseName: 'Squat', Sets: 5, Reps: 5 },
+        { exerciseName: 'Press', Sets: 5, Reps: 5 },
+        { exerciseName: 'Deadlift', Sets: 1, Reps: 5 },
+      ],
+    ],
+    notes: 'anal',
+  },
+};
+
+interface Exercise {
+  value: string;
+  text: string;
+}
+
+interface ExerciseLogEntry {
+  exerciseName: string;
+  Sets: number;
+  Reps: number;
+  Weight: number;
+}
+
+export function addExerciseToDatabase(exerciseName: string): void {
+  let currentDatabase: Array<Exercise>;
+  if (localStorage.getItem('exerciseDatabase') === null) {
+    currentDatabase = defaultExerciseDatabase;
+  } else {
+    currentDatabase = JSON.parse(localStorage.getItem('exerciseDatabase') as string);
+  }
+
+  currentDatabase.push({ value: exerciseName, text: exerciseName });
+  localStorage.setItem('exerciseDatabase', JSON.stringify(currentDatabase));
+}
+
+export function retrieveExerciseDatabase(): Array<Exercise> {
+  if (localStorage.getItem('exerciseDatabase') === null) {
+    return defaultExerciseDatabase;
+  }
+  return JSON.parse(localStorage.getItem('exerciseDatabase') as string);
+}
+
+export function addLogToDatabase(exerciseLog: Array<ExerciseLogEntry>, day: string): void {
```

Comment:
Remove "ToDatabase" from the names, this is already in the Database file, so if the function is called "addLog" it means it's adding to the db

Reply:
fixed